utils/SimpleJson.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <assert.h>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace utils
+{
+
+class SimpleJson
+{
+public:
+    enum Type
+    {
+        None,
+        Boolean,
+        Null,
+        String,
+        Integer,
+        Float,
+        Object,
+        Array
+    };
+    static SimpleJson create(const char* cursorIn, size_t length);
+    static const SimpleJson SimpleJsonNone;
+
+    SimpleJson() : _cursorIn(nullptr), _cursorOut(nullptr), _type(Type::None) {}
+    SimpleJson(const SimpleJson& ref)
+    {
+        _cursorIn = ref._cursorIn;
+        _cursorOut = ref._cursorOut;
+        _type = ref._type;
+    }
+    SimpleJson& operator=(const SimpleJson& ref)
+    {
+        _cursorIn = ref._cursorIn;
+        _cursorOut = ref._cursorOut;
+        _type = ref._type;
+        return *this;
+    }
+
+    Type getType() const { return _type; }
```

Comment:
I wonder if this variant like typing is benefitial. 
One thing I lack here is typed JsonArray that has begin end methods so I can iterate over the array items.
Putting that into a generic SimpleJson seems a bit akwars as the object is usually not an array.

Reply:
Removed. Changed to per-type getters returnong Optional<>